server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/vacation/VacationResponseSet.scala

```diff
@@ -71,7 +79,15 @@ case class VacationResponsePatchObject(jsObject: JsObject) {
       .build()
 
     maybeError.map(e => Left(e))
-      .getOrElse(Right(patch))
+      .getOrElse(validatePatch(patch))
+  }
+
+  private def validatePatch(patch: VacationPatch): Either[IllegalArgumentException, VacationPatch] = {
+    if(patch.getFromDate.isModified && patch.getToDate.isModified && patch.getFromDate.get().isAfter(patch.getToDate.get())) {
+      Left(new IllegalArgumentException("fromDate must be after toDate"))
```

Comment:
date are inverted in the error message than in the condition.
a.isAfter(b) return true if a is more recent than b.
Please check what you really want and use the more comprehensible 'more recent than' or 'older than'